r/R/parquet.R

```diff
@@ -567,3 +568,14 @@ ParquetArrowReaderProperties <- R6Class("ParquetArrowReaderProperties",
 ParquetArrowReaderProperties$create <- function(use_threads = option_use_threads()) {
   parquet___arrow___ArrowReaderProperties__Make(isTRUE(use_threads))
 }
+
+with_safe_locale <- function(expr) {
+  # ARROW-7288: due to a bug in libstdc++ on mingw, std::regex can blow up
+  # on other locales, and the parquet reader calls it
+  if (tolower(Sys.info()[["sysname"]]) == "windows") {
+    old_locale <- Sys.getlocale("LC_COLLATE")
+    Sys.setlocale("LC_COLLATE", "C")
+    on.exit(Sys.setlocale(old_locale))
```

Comment:
```suggestion
    on.exit(Sys.setlocale("LC_COLLATE", old_locale))
```

`category` is needed here, as the error message on CI suggests.

```
Error: Error: invalid 'category' argument
```